Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -1610,14 +1610,19 @@ bool LocalDOMWindow::consumeTransientActivation()
         RefPtr localFrame = dynamicDowncast<LocalFrame>(frame.get());
         if (!localFrame)
             continue;
-        auto* window = localFrame->window();
-        if (!window || window->lastActivationTimestamp() != MonotonicTime::infinity())
-            window->setLastActivationTimestamp(-MonotonicTime::infinity());
+        if (auto* window = localFrame->window())
+            window->consumeLastActivationIfNecessary();
     }
 
     return true;
 }
 
+void LocalDOMWindow::consumeLastActivationIfNecessary()
+{
+    if (m_lastActivationTimestamp != MonotonicTime::infinity())
```

Comment:
nit: may be nice to use this since we're touching this code:
```
if (!std::isinf(m_lastActivationTimestamp))
```

We support this nowadays:
```
inline bool isinf(WTF::MonotonicTime time)
{
    return std::isinf(time.secondsSinceEpoch().value());
}
```